src/game.cpp

```diff
@@ -5963,8 +5967,9 @@ void game::print_items_info( const tripoint &lp, const catacurses::window &w_loo
                                 pgettext( "%s is the name of the item.  %d is the quantity of that item.", "%s [%d]" ),
                                 it.first.c_str(), it.second );
             } else {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white, it.first );
+                trim_and_print( w_look, point( column, ++line ), max_width, obj_color[counter], it.first );
```

Comment:
This will not display the correct color since the item names and item colors are stored in the different order in `obj_color` (which is a vector) and `item_names` (which is a map). You can instead change `item_names`'s type to `std::map<std::string, std::pair<int, nc_color>>` and store both the count and the color in the same map, so the order is always the same.

Reply:
Actually there's also the problem of items of the same type having different colors in the inventory (unrotten and rotten food, for example), so the map should probably better be indexed by both the item id and color, with item count as the mapped value.

Reply:
could you show me a bit please and make code suggestion ?

Reply:
yes, i confirm, the colors are fine in vanilla. what you see in inventory, you see the same on look_around. if you still think the way i did it is not optimal, i would be glad if you could submit your suggestion. i'd be happy to study it.

Reply:
Actually I think the colors in your screenshot are incorrect. For example, the `plastic bag of shelled peanuts (fresh)` should be the one with the blue color, but in your screenshot it is the `joint` which has the blue color.

Reply:
you were right, the colors are shifted. i will try to do as you said.
if you could help a bit it would be even better.

if i make `std::map<std::string, std::pair<int, nc_color>> item_names;`
how do i store and access the different elements in that ?
this is wrong and it's all i could think of: 
```
std::map<std::string, std::pair<int, nc_color>> item_names;
for( auto &item : m.i_at( lp ) ) {
    ++item_names[item.tname()].first;
    item_names[item.color_in_inventory()].second;
}
```

Reply:
I think it can be
```
std::map<std::string, std::pair<int, nc_color>> item_names;
for( auto &item : m.i_at( lp ) ) {
    ++item_names[item.tname()].first;
    item_names[item.tname()].second = item.color_in_inventory();
}
```
Of course this doesn't take into account the fact that two items with the same name could have different colors (I can't think of an example for now, but nothing is stopping it from happening in the future), but it's a good starter for now.

Reply:
thanks a lot for the advice Qrox, i will try that now.
i have a question, i noticed that previously, my false colors were consistently wrong in the same way, i mean the colors were shifted by one or two. would it be considered bad practice if i would have just tried to offset my counter by the correct amount ?

Reply:
They may be consistently wrong for the exact stack of items, but not if a different stack of items is involved. So no, do not just offset the counter.